train.py

```diff
@@ -157,7 +178,21 @@ def train(config):
     config.argv_update(left_argv)
 
     config.exp_name = basename(args.config).split(".")[0]
-    config.exp_version = datetime.datetime.now().strftime("%Y%m%d_%H%M%S") if not args.exp_version else args.exp_version
+    config.exp_version = (
+        datetime.datetime.now().strftime("%Y%m%d_%H%M%S")
+        if not args.exp_version
+        else args.exp_version
+    )
+    if config.get("wandb", False):
+        run = wandb.init()
 
-    save_config_file(config, Path(config.result_path) / config.exp_name / config.exp_version)
+    save_config_file(
+        config, Path(config.result_path) / config.exp_name / config.exp_version
+    )
     train(config)
+
+    if config.get("wandb", False):
```

Comment:
IMHO this if will look better as last part of `train` function body, so please move this code block.